src/dune

```diff
@@ -19,7 +19,32 @@
  (modes byte native)
  (public_name mtt)
  (wrapped true)
- (libraries base stdio sexplib sedlex menhirLib pprint)
+ (libraries base stdio sexplib sedlex menhirLib pprint zarith)
  (preprocess
   (pps ppx_let ppx_compare ppx_sexp_conv ppx_string_interpolation sedlex.ppx))
  (synopsis "Modal Type Theory Library"))
+
+;; Rules for generation a fresh ParserErrors.messages file
+
+(rule
+ (targets NewParserErrors.messages)
+ (deps Parser.mly)
+ (action
+  (with-stdout-to
+   NewParserErrors.messages
+   (run %{bin:menhir} --list-errors Parser.mly))))
+
+(rule
+ (targets UpdParserErrors.messages)
+ (deps NewParserErrors.messages Parser.mly)
+ (action
+  (with-stdout-to
+   UpdParserErrors.messages
+   (run %{bin:menhir} --merge-errors ParserErrors.messages --merge-errors
+     NewParserErrors.messages Parser.mly))))
+
+(rule
+ (alias runtest)
+ (deps ParserErrors.messages UpdParserErrors.messages)
+ (action
+  (diff ParserErrors.messages UpdParserErrors.messages)))
```

Comment:
This part somehow creates some issues for the recursion PR -- they see Menhir's output after `dune runtest`. We should probably move some commands to the `ParserErrors.ml` rule at the beginning of this file:

```
(rule
 (targets ParserErrors.ml)
 (deps ParserErrors.messages Parser.mly)
 (action
  (with-stdout-to
   ParserErrors.ml
   (run %{bin:menhir} --compile-errors ParserErrors.messages Parser.mly))))
```

Reply:
I added `UpdParserErrors.messages` and `NewParserErrors.messages` to deps. After this change and commit you suggested below, `Recursion` branch builds fine